javascript/ql/experimental/adaptivethreatmodeling/lib/experimental/adaptivethreatmodeling/ATMConfig.qll

```diff
@@ -154,6 +154,17 @@ abstract class AtmConfig extends JS::TaintTracking::Configuration {
     not AtmResultsInfo::isFlowLikelyInBaseQuery(source.getNode(), sink.getNode()) and
     score = AtmResultsInfo::getScoreForFlow(source.getNode(), sink.getNode())
   }
+
+  /**
+   * Holds if if `sink` is an effective sink with flow from `source` which gets used as a sink candidate for scoring
+   * with the ML model.
+   */
+  predicate isSinkCandidate(JS::DataFlow::PathNode sink) {
+    exists(JS::DataFlow::PathNode source |
+      this.hasFlowPath(source, sink) and
```

Comment:
```suggestion
      this.hasFlow(source, sink) and
```

Reply:
I think this change is incorrect. We need to use `hasFlowPath` here, just like we use it in `hasBoostedFlowPath` above, so that the test is as close as possible to the actual extraction queries.

Reply:
I'll have to look into what the difference is on Monday morning. I think it's correct, but that's not good enough if you disagree.

Reply:
The question is what's the harm in leaving both this and `PathNode` as they are, so that they match the extraction queries as closely as possible. If there's no major downside, I think that's the right call. 

Ideally the tests would call the extraction queries directly, to guarantee we're testing exactly what we intend to test, but in this case we can't do that because the extraction queries use the ML model. So the idea was to do everything exactly like the extraction queries other than producing a score. I understand we may be able to use `Node` rather than `PathNode` and `hasFlow` rather than `hasFlowPath`, but I don't understand what benefit we'd get from that that's more important than ensuring the test matches the extraction queries as closely as possible.

Reply:
I thought about this some more over dinner and came to the conclusion that it's probably not worth holding this up over something that is very likely only a tiny performance issue. 👍

Reply:
...so I walked upstairs to tell you that. I'll now look at the discussions you and Jean had in this PR and if there's nothing left open from him, will approve.